File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        vector<int> seen(130);
        int res = 0, i = 0;
        for(int j = 0; j < s.length(); ++j) {
            while(seen[s[j]] and i < j) {
                --seen[s[i++]];
            }
            ++seen[s[j]];
            int len = j - i + 1;
            res = max(len, res);
        }
        return res;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        res, i, seen = 0, 0, [0] * 130
        for j in range(len(s)):
            while seen[ord(s[j])] and i < j:
                seen[ord(s[i])] -= 1
                i += 1
            seen[ord(s[j])] += 1
            length = j - i + 1
            res = max(length, res)
        return res